hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -599,6 +586,9 @@ def generate_new_examples(self):
             return
 
         zero_data = self.cached_test_function(hbytes(self.settings.buffer_size))
+        if zero_data.status > Status.OVERRUN:
+            self.__data_cache.pin(zero_data.buffer)
```

Comment:
I assume the idea here is that the zero-buffer result is so common and useful that it's never worth evicting, even if it goes temporarily unused for a long series of shrink attempts.

Reply:
Yeah. Also that we definitely know we're going to use it later during shrinking if we find an interesting example.